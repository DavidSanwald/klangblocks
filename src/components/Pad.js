import { observer, inject } from 'mobx-react'
import {observable, action, autorun} from 'mobx'
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import sample from 'lodash.sample'
import StyledPad from './StyledPad'



@inject('mainStore')
@observer
class Pad extends Component {
  constructor (props) {
    super(props)
    this.id = props.id
}



  handleClick = () => {
    this.props.pad.toggleSelected()
  };

  findPad = (pads) => {
    return pads.id === this.props.id
  };

  render () {
    const actualPad = this.props.mainStore.pads.find(this.findPad)
    let test = this.props.mainStore.isRunning




    return (
      <StyledPad onClick={this.handleClick}
        isPlaying={actualPad.isPlaying}
        isActive={actualPad.isSelected}
       />
    )
  }
}

Pad.propTypes = {
  id: PropTypes.string.isRequired,
  key: PropTypes.string.isRequired
}

export default Pad
