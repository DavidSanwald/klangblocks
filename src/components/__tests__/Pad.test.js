import React from 'react'
import renderer from 'react-test-renderer'
import Pad from '../Pad'

const spyOnClick = jest.fn();

describe("Pad", function() {
  let props = {}
  let pad = {}
  beforeEach(function() {

    props = {id: "111"}
    this.store = {
      pads: [
        {state: "idle", id: "111", m: 1, n: 1},
        {state: "idle", id: "222", m: 1, n: 1},
        {state: "idle", id: "222", m: 1, n: 1},
      ]
    }
  })

  it('renders without crashing', function() {
    const tree = renderer.create(<Pad.wrappedComponent
    mainStore={this.store} id={props.id}  state='playing' onClick={spyOnClick} pad={pad} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
