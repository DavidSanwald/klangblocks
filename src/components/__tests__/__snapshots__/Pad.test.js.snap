// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pad renders without crashing 1`] = `
<rect
  className="bmVkCM"
  height="10.3"
  onClick={[Function]}
  width="10.3"
  x={10}
  y={10}
/>
`;
